subprojects/core/src/main/java/org/gradle/execution/commandline/CommandLineTaskConfigurer.java

```diff
@@ -94,4 +98,18 @@ private List<String> configureTasksNow(Collection<Task> tasks, List<String> argu
         }
         return remainingArguments;
     }
+
+    private void mutuallyExcludeOppositeOptions(CommandLineParser parser, List<OptionDescriptor> validCommandLineOptions) {
```

Comment:
:art: `BooleanOptionElement` should not reference it's opposite.  IIUC this function is the only consumer of that.  `CommandLineTaskConfigurer` should not know how to calculate mutual exclusivity between boolean opposites. What I would do instead is have `TaskOptionSupplier` provide the right data ready for `CommandLineTaskConfigurer` to declare `allowOneOf` without doing any calculations. Then the whole concept of boolean opposites is encapsulated in `TaskOptionSupplier`.

This will further simplify things.